joomla/joomla-cms
# [4.1][regression] Correctly handle save2copy for menus

Pull Request for Issue #37812

### Summary of Changes

Restore the existing behavior before the regression caused by https://github.com/joomla/joomla-cms/pull/37395 as well as fix the Issue from the original report

### Testing Instructions

- setup a 4.1.2 site
- create 2 menus
- create one menu items for the first menu
- open that menu item in edit mode
- change the menu setting to the seccond menu
- Use the "save2copy" feature.
- the change to the menu setting is ignored
- update to 4.1.3
- repeat the steps from above
- it works
- but try it without changing the assigned menu
- it fails as the alias exists already
- apply the patch
- confirm that both cases still work as expected
- try to only change the menu item title
- save2copy
- confirm that the alias number is not just increased but followed the change to the menu title
- try to only change the alias but keep the title
- save2copy
- confirm that the alias has been updated but not the title.

### Actual result BEFORE applying this Pull Request

Issues of "alias exists already" while with save2copy we have an "increment the ID" feature

### Expected result AFTER applying this Pull Request

The alias will be updated as expected any by using save2copy you should not get alias dubliceted issues.

### Documentation Changes Required

n.a

### Mentions

@toivo 
@bembelimen 

Involved within the PR which caused the regression:
@anmode
@chmst
@richard67
@vorayash
@Quy
